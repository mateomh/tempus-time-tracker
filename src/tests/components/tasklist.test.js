import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import TaskList from '../../components/tasklist';
import rootReducer from '../../reducers/root';

describe('TaskList Component', () => {
  let mockStore;

  beforeEach(() => {
    const tasks = [
      {
        desc: 'Test 1',
        time: 123,
        created_at: 1,
        id: 1,
      },
      {
        desc: 'Test 2',
        time: 456,
        created_at: 2,
        id: 2,
      },
    ];

    const mockState = {
      sessionUser: {},
      filter: {},
      categories: [],
      tasks,
      total: 0,
      error: '',
    };

    mockStore = createStore(rootReducer, mockState);
  });

  it(
    '1. Renders correctly',
    () => {
      render(
        <Provider store={mockStore}>
          <TaskList />
        </Provider>,
      );
      let item = screen.getByText('Test 1');
      expect(item).toBeDefined();
      item = screen.getByText('Test 2');
      expect(item).toBeDefined();
      item = screen.getByText('123');
      expect(item).toBeDefined();
      item = screen.getByText('456');
      expect(item).toBeDefined();
    },
  );

  it(
    '2. Checks it renders correctly against the snapshot',
    () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <TaskList />
          </Provider>,
        ).toJSON();
      expect(tree).toMatchSnapshot();
    },
  );
});
