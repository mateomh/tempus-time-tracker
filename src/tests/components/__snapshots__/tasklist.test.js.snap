// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Component 2. Checks it renders correctly against the snapshot 1`] = `
<div
  className="List"
>
  <div
    className="Item"
  >
    <p
      className="Desc"
    >
      <b>
        Test 1
      </b>
      <small
        className="Small"
      >
        Created on:  
        31-Dec-1969
      </small>
    </p>
    <p
      className="Time"
    >
      <b>
        123
      </b>
       hours
    </p>
  </div>
  <div
    className="Item"
  >
    <p
      className="Desc"
    >
      <b>
        Test 2
      </b>
      <small
        className="Small"
      >
        Created on:  
        31-Dec-1969
      </small>
    </p>
    <p
      className="Time"
    >
      <b>
        456
      </b>
       hours
    </p>
  </div>
  <button
    className="Button"
    onClick={[Function]}
    type="button"
  >
    New task for 
  </button>
</div>
`;
